pages/voting-guides/index.tsx

```diff
@@ -0,0 +1,149 @@
+import { useState, useEffect } from "react";
+import { NextPage } from "next";
+import Head from "next/head";
+import { 
+  Scalars,
+  useUpsertVotingGuideMutation,
+  useVotingGuidesByUserIdQuery,
+  VotingGuideResult } from "generated";
+import { Layout, Avatar, FlagSection, Button } from "components";
+import { useAuth } from "hooks/useAuth";
+import styles from "./VotingGuides.module.scss";
+
+
+const VotingGuideCard = ({ guide }:{ guide: VotingGuideResult }) => {
+  const {user} = guide;
+  const {firstName, lastName, avatarUrl, username} = user;
+  const name = firstName ? `${firstName} ${!!lastName ? lastName:''}`: username;
+  return(
+  <div className={`${styles.roundedCard} ${styles.votingCardContainer}`}>
+  <div className={styles.guideContainer}>
+    <div className={styles.avatarContainer}>
+      <Avatar src={avatarUrl || ""} size={80} fallbackSrc="https://www.gravatar.com/avatar/" alt={name} />
+      <h4>{name}</h4>
+    </div>
+    <div className={styles.buttonWrapper}>
+      <Button large primary theme="blue" label="Edit">Edit</Button>
+      <span className={styles.shareButton}>
+        <Button large primary theme="yellow" label="Share">Share</Button>
+      </span>
+    </div>
+  </div>
+</div>);
+};
+
+interface VotingGuideElection {
+  date: string,
+  description?: string,
+  guides: VotingGuideResult[],  
+  id: Scalars['ID'],
+  title: string,
+}
+
+const getElectionsFromGuides = (guides: VotingGuideResult[]): VotingGuideElection[]=>{
+  if (guides){
+    return guides.reduce<VotingGuideElection[]>((result, value) => {
+      const i = result.findIndex(v=>v.id===value.electionId);
+      if (i<0) {
+        const electionInfo = {
+          id: value.electionId,
+          title: value.election.title,
+          description: value.election.description,
+          date: value.election.electionDate,
+          guides: [value]
+        } as VotingGuideElection;
+        result.push(electionInfo);
+      } else {
+        result[i]?.guides.push(value);
+      }
+      return result;
+    }, [] as VotingGuideElection[]);
+  } else return [] as VotingGuideElection[];
+};
```

Comment:
I see what you were trying to do here, but I think that this won't be relevant when we get closer to the mocks.  A voting guide will be a layer on top of the election, so you won't have to derive this

Reply:
Henry's design shows Elections with lists of associated guides. This is the reasoning for this. as an alternative we could have a query for the current user than returned a list of elections with associated guides. I wrote this function to transform and sort the list that way, but obviously would be better to take that logic out of the client.

Reply:
cool i've asked henry some questions about this voting guide list view but I'll plan to create a resolver that returns something like: 
```rust
pub struct ElectionVotingGuides {
    pub election: ElectionResult,
    pub voting_guides: Vec<VotingGuideResult>,
    // potentially other data that could be useful on this page
}
````

Reply:
That would be great!